automation_tools/satellite6/upgrade/__init__.py

```diff
@@ -214,7 +214,7 @@ def product_upgrade(product):
                                 upgraded = str(execute(
                                     get_sat_cap_version, 'cap', host=cap_host
                                     )[cap_host])
-                                if current:
+                                if current != 'None':
```

Comment:
I think the `get_sat_cap_version` should raise an exception when the version can't be figured out since that knowing the version is required. Doing that cast, for None as string seems a bad code smell to me.